src/GafferScene/DeleteFaces.cpp

```diff
@@ -0,0 +1,141 @@
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "boost/format.hpp"
+
+#include "IECore/MeshPrimitive.h"
+#include "IECore/MeshAlgo.h"
+
+#include "Gaffer/StringPlug.h"
+
+#include "GafferScene/DeleteFaces.h"
+
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferScene;
+
+IE_CORE_DEFINERUNTIMETYPED( DeleteFaces );
+
+size_t DeleteFaces::g_firstPlugIndex = 0;
+
+DeleteFaces::DeleteFaces( const std::string &name ) : SceneElementProcessor( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	addChild( new StringPlug( "delete", Plug::In, "delete" ) );
+
+	// Fast pass-through for things we don't modify
+	outPlug()->attributesPlug()->setInput( inPlug()->attributesPlug() );
+	outPlug()->transformPlug()->setInput( inPlug()->transformPlug() );
+}
+
+DeleteFaces::~DeleteFaces()
+{
+}
+
+Gaffer::StringPlug *DeleteFaces::deletePlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex );
+}
+
+const Gaffer::StringPlug *DeleteFaces::deletePlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex );
+}
+
+void DeleteFaces::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	SceneElementProcessor::affects( input, outputs );
+
+	if( input == deletePlug() )
+	{
+		outputs.push_back( outPlug()->objectPlug() );
+	}
+	else if( input == outPlug()->objectPlug() )
+	{
+		outputs.push_back( outPlug()->boundPlug() );
+	}
+}
+
+bool DeleteFaces::processesBound() const
+{
+	return true;
+}
+
+void DeleteFaces::hashProcessedBound( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+
+	hashProcessedObject( path, context, h );
+}
+
+Imath::Box3f DeleteFaces::computeProcessedBound( const ScenePath &path, const Gaffer::Context *context, const Imath::Box3f &inputBound ) const
+{
+	ConstObjectPtr object = outPlug()->objectPlug()->getValue();
+	if( const Primitive *primitive = runTimeCast<const Primitive>( object.get() ) )
+	{
+		return primitive->bound();
+	}
+	return inputBound;
+}
+
+bool DeleteFaces::processesObject() const
+{
+	return true;
+}
+
+void DeleteFaces::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	deletePlug()->hash( h );
+}
+
+IECore::ConstObjectPtr DeleteFaces::computeProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::ConstObjectPtr inputObject ) const
+{
+	const MeshPrimitive *mesh = runTimeCast<const MeshPrimitive>( inputObject.get() );
+	if( !mesh )
+	{
+		return inputObject;
+	}
+
+	std::string deletePrimVarName = deletePlug()->getValue();
```

Comment:
Let's add an early out if the name is empty, rather than have it throw later on :

```
if( deletePrimVarName.empty() )
{
    return inputObject;
}
```